src/internal/coredb/projects.go

```diff
@@ -3,13 +3,15 @@ package coredb
 import (
 	"context"
 	"fmt"
+	"io"
+	"time"
+
 	"github.com/gogo/protobuf/types"
 	"github.com/jmoiron/sqlx"
+
 	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
 	"github.com/pachyderm/pachyderm/v2/src/internal/pachsql"
 	"github.com/pachyderm/pachyderm/v2/src/pfs"
-	"io"
-	"time"
 )
 
 type ProjectIterator struct {
```

Comment:
This kind of iterator will block the connection while it iterates.  I can't see everywhere this is used from the current diff, but this isn't the kind of iterator we would want to use for streaming out to an RPC.  For that it should have a `*sqlx.DB` internally, and buffer a page of results at a time, making queries as needed.

Additionally, if you implement `stream.Iterator` (Next takes a context) then you can use stream.ForEach and stream.Collect whereever this is used.  This isn't in the diff, so not a blocker for this PR.